DeratControl.Domain/Entities/Facility.cs

```diff
@@ -17,5 +19,21 @@ private Facility()
             this.Perimeters = new List<Perimeter>();
             this.Reviews = new List<Review>();
         }
+
+        public void RemovePerimeter(PerimeterType perimeterType)
+        {
+            var perimeterExists = this.Perimeters.Any(f => f.PerimeterType == perimeterType);
```

Comment:
I think we should check if perimeter has active reviews before we can delete it

Reply:
You're doing double work here
First you loop through the Perimeters collection to find any element that satisfies your predicate
But after you loop through the collection again,

Basically you can use the second query
but call not .Single() but .SingleOrDefault()

In case you didn't find a perimeter you'll get null
if null throw an exception otherwise pass this object to  this.Perimeters.Remove();